designer/client/src/components/graph/node-modal/fragment-input-definition/settings/variants/fields/ValidationFields.tsx

```diff
@@ -25,11 +26,20 @@ export default function ValidationFields({
     readOnly,
     errors,
     name,
+    typ,
 }: ValidationFields) {
     const { t } = useTranslation();
 
     const validationExpressionFieldName: FieldName = `$param.${name}.$validationExpression`;
 
+    const extendedVariableType = useMemo(
+        () => ({
+            ...variableTypes,
+            value: { refClazzName: typ.refClazzName, type: "TypedClass", params: [] } as ReturnedType,
```

Comment:
This request template is not always correct. When you select a parameter type with a generic parameter like `List[Unknown]` the response for /suggestions is `Type's generic parameters don't match expected type parameters: found List[], expected List[E]. This may be caused by passing incorrect arguments or using type aliases.` For these types, FE should pass these type parameters in the `params` array. 

You can see which types I mean when you check the `processDefinitionData` response in the `typesInformation` array.

I think if BE provides refClazzName then it should provide info about its parameters. But without changing BE I guess you could try to get that info from the `typesInformation` request (I'm guessing these class names are unique and it should work?)

Reply:
You are right, I changed it to  use data from `typesInformation` ✔️